donor-portal/src/pages/home/index.tsx

```diff
@@ -74,23 +77,46 @@ const packageList = [
   },
 ];
 
+const pledgedPackages = [
+  {
+    packageID: 2,
+    name: "Medical Supplies | Anuradpura",
+    hospital: "General Hospital : Fight for their Lives in Anuradapura",
+    description:
+      "Kole Intellectual Forum intends to influence the ideas of policy makers at all levels, educationists, community and youth in Uganda to inclu... read more",
+    totalAmount: 150000,
+    pledgedPercentage: 20,
+  },
+];
+
 export function Home() {
-  const [aidPackages, setAidPackages] = useState<Array<DonarAidPackage>>();
-  const [goalFilter, setGoalFilter] = useState<GoalStatus>(
-    GoalStatus.GOAL_PENDING
+  const userId = useContext(UserContext);
+  const [goalPendingAidPackages, setGoalPendingAidPackages] = useState<
+    DonorAidPackage[]
+  >([]);
+  const [alreadyPledgedAidPackages, setAlreadyPledgedAidPackages] = useState<
+    DonorAidPackage[]
+  >([]);
+  const [goalReachedAidPackages, setGoalReachedAidPackages] = useState<
+    DonorAidPackage[]
+  >([]);
+  const [activeTabItem, setActiveTabItem] = useState<TabItems>(
+    TabItems.GOAL_PENDING
   );
 
   useEffect(() => {
-    setAidPackages(
-      packageList.filter((donorPackage) => {
-        if (goalFilter === GoalStatus.GOAL_PENDING) {
-          return donorPackage.pledgedPercentage !== 100;
-        } else {
-          return donorPackage.pledgedPercentage === 100;
-        }
-      })
+    const aidPackages = packageList;
+    setGoalReachedAidPackages(
+      aidPackages.filter((aidPackage) => aidPackage.pledgedPercentage == 100)
+    );
+    setGoalPendingAidPackages(
+      aidPackages.filter((aidPackage) => aidPackage.pledgedPercentage != 100)
     );
```

Comment:
Move this part to a seperate effect as it doesn't depend on the `userId`

Reply:
Sure

Reply:
Fixed with the commit 751be2d, Please review